### PR TITLE
[DO NOT MERGE] Allow small images to served as fallback

### PR DESCRIPTION
**Pull request by jfsoul**
## What does this change?
~~This PR selects the smallest _'large enough'_ image to be served when our image resizing service is switched off.  The logic to select the largest image when image resizing is enabled was already in place, but didn't work because `elementFor` selected the largest image in every case.~~
This PR selects the _closest_ image size to the required width when our image resizing service is switched off

A change was needed for trail images as well because the pressed json only contained the largest 5:3 ratio image for fronts.  Now we supply the largest and smallest _'large enough'_ 5:3 image plus the master (if 5:3) for facia to select the correct choice at render time.  If this is to be merged at any point I think we should get composer to add another thumbnail size (1000px) and press that size too.

The logic needs cleaning up still, but just wanted to get a feel for the viability of this solution.

## What is the value of this and can you measure success?
An imgix fallback using sizes from the grid could be viable with this.  ~~It needs more testing of page weights, but early testing of pressed fronts in DEV showed only about a 10% increase in page weight with no effect on pressed size.~~ Page weight on a network front increased about 20-30% as a result of this, not including front image overrides.

## Does this affect other platforms - Amp, Apps, etc?
No

## Tested in CODE?
Will be doing so

### CODE REVIEW

File: common/app/views/support/Profile.scala [hunk 1]
```diff
@@ -21,7 +21,7 @@ sealed trait ElementProfile {
   def autoFormat: Boolean
 
   def elementFor(image: ImageMedia): Option[ImageAsset] = {
-    val sortedCrops = image.imageCrops.sortBy(-_.width)
```
Comment (TBonnin) [inline common/app/views/support/Profile.scala]: That meant we are currently always serving the largest one. Right?

File: common/app/model/trails.scala [hunk 2]
```diff
@@ -24,24 +24,22 @@ object Trail {
       .orElse(elements.videos.headOption.map(_.images))
       .orElse(elements.thumbnail.map(_.images))
 
-    // Try to take the master 5:3 image. At render-time, the image resizing service will size the image according to card width.
-    // Filtering the list images here means that facia-press does not need to slim down the Trail object.
+    // Try to take the master 5:3 image, the largest and the smallest 5:3 image. At render-time, the image resizing service
+    // will size the largest image according to card width.  If the resizing service is switched off, we use the smallest.
```
Comment (paperboyo) [inline common/app/model/trails.scala]: > will size the largest image
should read
will size the master asset

(unless that's how it should always work for the resizer)

File: common/app/views/support/Profile.scala (hunk 1 above)
Comment (jfsoul) [inline common/app/views/support/Profile.scala]: Yep, whether the switch for resizing is on or off.

Comment (TBonnin) [inline common/app/views/support/Profile.scala]: bug or on purpose? 🙀

File: common/app/model/trails.scala (hunk 2 above)
Comment (jfsoul) [inline common/app/model/trails.scala]: I think the comment matches the functionality I have switched it to, but you're right I should prioritise the master as it does now - so I will clean that up before this gets close to being merged.

File: common/app/views/support/Profile.scala (hunk 1 above)
Comment (jfsoul) [inline common/app/views/support/Profile.scala]: I think intentional, possibly due to some usage I haven't considered yet. Will make sure to find out if we decide to go down this route.

File: common/app/model/trails.scala (hunk 2 above)
Comment (paperboyo) [inline common/app/model/trails.scala]: Thanks. I’m a master 👮 here. And [there](https://trello.com/c/cWSiXgfV/87-all-imagery-to-properly-use-master-assets-and-resizer-responsively).

If resizer -> **always** master.

**Comment by paperboyo** [comment]
This is a massive :gem:. A 20MB+ an-image-massive when actually needed :-)

> From the candidate images, choose the master 5:3 image, the largest 5:3 image and the smallest 5:3 image with a width >= 460. At render time we select from these based on whether using the resizing service.

I wouldn't know but if it doesn't cost us nothing why not to choose from all available media-service crops? There is a [static array of them](https://github.com/guardian/grid/blob/master/cropper/app/lib/Config.scala#L40) + the 1:1-sized one. This way the site with the fallback on could look better and weigh less, no?

File: common/app/views/support/Profile.scala (hunk 1 above)
Comment (paperboyo) [inline common/app/views/support/Profile.scala]: 🤔 (pls, let me know what you found out)

Comment (rich-nguyen) [inline common/app/views/support/Profile.scala]: I don't remember that being intentional, and I also think that the logic predates the two incarnations of image resizing we've used, plus the introduction of master images. So I'd say choose an approach for image selection (whilst resizing is disabled) which the team thinks works for you today (and tomorrow).

File: docs/06-features-and-components/02-trail-images.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # How are trail pictures picked in Frontend?
 
 Use the rules below like a filter to narrow down the content api response into a list of candidate images.
-From the candidate images, choose the master 5:3 image, or simply the largest 5:3 if a master wasn't found.
+From the candidate images, choose the master 5:3 image, the largest 5:3 image and the smallest 5:3 image with a width >= 460. At render time we select from these based on whether using the resizing service.
```
Comment (rich-nguyen) [inline docs/06-features-and-components/02-trail-images.md]: 👍

**Comment by jfsoul** [comment]
@paperboyo 

> I wouldn't know but if it doesn't cost us nothing why not to choose from all available media-service crops?

Fair point, but there is a cost in that all the pressed front data will increase in size.  I don't imagine it will be much larger, but we try to slim it down as much as possible and this fallback will be rarely used anyway :crossed_fingers:

**Comment by jfsoul** [comment]
This should also now resolve https://github.com/guardian/frontend/issues/11754

**Comment by jfsoul** [comment]
Closing this for now as I think the investigation is finished.  If this is to be resurrected here is the TODO:

- [ ] Ask composer to pass through another thumbnail size if we want better quality imagery for large facia cards (1000px) (optional)

- [ ] Press the extra thumbnail size by changing `trails.scala` (optional)

- [ ] Work out what to do with facia image overrides (replacements, cutouts and slideshows).  Do we ignore these and show the usual image, or just show a blank card in case the original image is inappropriate?

- [ ] Do we continue to show non-grid images?  We could avoid the risk of greater page weight by only serving from media.guim.co.uk, not static (old), uploads (cutouts).

**Comment by TBonnin** [comment]
That's great investigation. Let's get together to answer all those optional questions.

**Pull request closed without merging**